### PR TITLE
OSDOCS-11269: Re-work second half of the "Getting Started with ROSA" Tutorials to be HCP specific

### PR DESCRIPTION
**Pull request by jneczypor**
[OSDOCS-11269](https://issues.redhat.com//browse/OSDOCS-11269): Re-work second half of the "Getting Started with ROSA" Tutorials to be HCP specific

Version(s):
4.16+

Issue:
https://issues.redhat.com/browse/OSDOCS-11269

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by jneczypor** [comment]
Hi @yuwang-RH - Can I please get a QE review of this PR? Thanks!

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+----
++
+.Example input
++
+[source,terminal]
+----
+$ rosa edit machinepool -c my-rosa-cluster --enable-autoscaling Default --min-replicas=2 --max-replicas=4
+----
++
+The above command creates an autoscaler for the worker nodes that scales between 2 and 4 nodes depending on the resources.
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc:61]: We need to clarify what `autoscaler`. There is a concept of `cluster autoscaler` in rosa. But enable machinepool autoscaling won't create an autoscaler to it.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+[source,terminal]
+----
+$ rosa edit machinepool -c <cluster-name> --enable-autoscaling <machinepool-name> --min-replicas=<num> --max-replicas=<num>
+----
++
+.Example input
++
+[source,terminal]
+----
+$ rosa edit machinepool -c my-rosa-cluster --enable-autoscaling Default --min-replicas=2 --max-replicas=4
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc:58]: No `Default` machinepool anymore. 
For classic, it is `worker`
For HCP it is `workers` or `workers-x` like `workers-0`

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+----
+$ rosa list machinepools -c <cluster-name>
+----
++
+.Example output
++
+[source,terminal]
+----
+ID         AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS     TAINTS    AVAILABILITY ZONES
+Default    Yes          2-4       m5.xlarge                           us-east-1a
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc:84]: No `Default` machinepool anymore. Better to use latest rosacli to update.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-deleting.adoc [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+[WARNING]
+====
+This command is non-recoverable.
+====
+
+. The CLI prompts you to confirm that you want to delete the cluster. Press *y* and then *Enter*. The cluster and all its associated infrastructure will be deleted.
++
+[NOTE]
+====
+All AWS STS and IAM roles and policies will remain and must be deleted manually once the cluster deletion is complete by following the steps below.
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-deleting.adoc:39]: @yingzhanredhat Can you please confirm this part? If HCP cluster deletion mentioned `the policies will remain, follow xxx to delete xxx` then let's file a bug. HCP is using managed policies those cannot be deleted

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+----
+rosa list machinepools --cluster=<cluster-name>
+----
++
+.Example output
++
+[source,terminal]
+----
+ID          AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS            TAINTS    AVAILABILITY ZONES
+Default     No           2         m5.xlarge                                  us-east-1a
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc:77]: No `Default` machine pool.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+[source,terminal]
+----
+$ rosa describe cluster --cluster=my-rosa-cluster | grep Compute
+----
++
+.Example output
++
+[source,terminal]
+----
+- Compute:                 3 (m5.xlarge)
```
Comment (xueli181114) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc:157]: The output is too old. And the machine type won't show.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+== Enabling autoscaling for an existing machine pool using the UI
+
+[NOTE]
+====
+Cluster autoscaling can be enabled at cluster creation by checking the *Enable autoscaling* checkbox when creating machine pools.
+====
+
+. Go to the *Machine pools* tab and click the three dots in the right..
+. Click *Scale*, then *Enable autoscaling*.
```
Comment (jmekkatt) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc:71]: It should be "edit" and no scale option available in UI.
Screenshot from 2024-09-24 11-48-58

Comment (jmekkatt) [reply rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-autoscaling.adoc:71]: After enable autoscaling option, user could update the minimum node count and maximum node count or leave the default values in both fields.
Finally user has to save the definition to take effect of changes in machine pool.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+ID          AUTOSCALING  REPLICAS  INSTANCE TYPE  LABELS            TAINTS    AVAILABILITY ZONES
+Default     No           2         m5.xlarge                                  us-east-1a
+----
+
+=== Creating a machine pool with the UI
+. Log in to the {cluster-manager-url} and click your cluster.
++
+image::cloud-experts-getting-started-managing-ocm-cluster.png[]
+
+. Click *Machine pools*.
```
Comment (jmekkatt) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc:85]: "Click Machine pools tab" would be more accurate.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
++
+image:cloud-experts-getting-started-managing-mp-ocm.png[]
+
+. Click *Add machine pool*.
+
+. Enter the desired configuration.
```
Comment (jmekkatt) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc:91]: User needs to click "Add machine pool" button to add a new machine pool at the end. Please add the same.

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-upgrading.adoc [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+. *Optional:* Schedule the cluster to upgrade at a later date and time by running the following command:
++
+[source,terminal]
+----
+$ rosa upgrade cluster -c <cluster-name> --version <desired-version> --schedule-date <future-date-for-update> --schedule-time <future-time-for-update>
+----
+
+== Manually upgrading your cluster using the UI
+. Log in to the OpenShift Cluster Manager, and select the cluster you want to upgrade.
+. Click *Settings*.
```
Comment (jmekkatt) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-upgrading.adoc:74]: Suggest to change Click Settings tab

**Comment by jmekkatt** [comment]
Hi @jneczypor , Added comments on UI specific doc changes. Kindly take a look. Thanks

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-support.adoc [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
++
+image::obtain-support-case.png[]
+
+. Log in to your Red{nbsp}Hat account.
+. Select the reason for contacting support.
++
+image::obtain-support-reason.png[]
+
+. Select *Red{nbsp}Hat OpenShift Service on AWS*.
+
```
Comment (EricPonvelle) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-support.adoc:40]: You need a `+` on this line to get the image and numbers to order correctly

image

**Review by EricPonvelle — COMMENTED** [review]
Two small fixes

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-deleting.adoc (hunk 4 above)
Comment (yingzhanredhat) [reply rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-deleting.adoc:39]: @xueli181114 It doesn't show this message. It just provide the operator-roles/oidc-provider deleted steps

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-upgrading.adoc [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+----
+$ rosa upgrade cluster -c <cluster-name> --version <desired-version>
+----
+endif::openshift-rosa-hcp[]
+ifdef::openshift-rosa-hcp[]
+. Schedule the cluster to upgrade within the hour by running the following command:
++
+[source,terminal]
+----
+$ rosa upgrade cluster -c --control-plane <cluster-name> --version <desired-version>
```
Comment (yingzhanredhat) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-upgrading.adoc:61]: it should be

> rosa upgrade cluster -c <cluster-name> --control-plane  --version <desired-version>

File: rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc [hunk 13]
```diff
@@ -0,0 +1,235 @@
...
+----
+$ rosa create machinepool --cluster=<cluster-name> --name=<machinepool-name> --replicas=<number-nodes>
+----
++
+.Example input
++
+[source,terminal]
+----
+ $ rosa create machinepool --cluster=my-rosa-cluster --name=new-mp
+ --replicas=2
```
Comment (yingzhanredhat) [inline rosa_learning/creating_cluster_workshop/cloud-experts-getting-started-managing-worker-nodes.adoc:31]: This flag should be put in previous line

> --replicas=2

**Pull request merged**